.github/workflows/test.yml

```diff
@@ -55,29 +53,71 @@ jobs:
           #   config: { suffix: -debug, bazel-args: --config=windows_dbg }
         exclude:
           # Skip the matrix generated Windows non-debug config to use the one added above.
-          - os:     { name : windows, image : windows-2025 }
+          - os: { name: windows, image: windows-2025 }
             config: { suffix: '' }
-          - os:     { name : windows, image : windows-2025 }
+          - os: { name: windows, image: windows-2025 }
             config: { suffix: -debug }
           # due to resource constraints, exclude the macOS-debug runner for now. linux-debug and
           # linux-asan should provide sufficient coverage for building in the debug configuration.
-          - os:     { name : macOS, image : macos-15 }
+          - os: { name: macOS, image: macos-15 }
             config: { suffix: -debug }
       fail-fast: false
     uses: ./.github/workflows/_bazel.yml
     with:
       image: ${{ matrix.os.image }}
       os_name: ${{ matrix.os.name }}
       suffix: ${{ matrix.config.suffix }}
-      extra_bazel_args: "--config=ci-test"
+      extra_bazel_args: '--config=ci-test'
     secrets:
       BAZEL_CACHE_KEY: ${{ secrets.BAZEL_CACHE_KEY }}
       WORKERS_MIRROR_URL: ${{ secrets.WORKERS_MIRROR_URL }}
   lint:
     uses: ./.github/workflows/_bazel.yml
     with:
-      extra_bazel_args: "--config=lint --config=ci-test"
+      extra_bazel_args: '--config=lint --config=ci-test'
       run_tests: false
     secrets:
       BAZEL_CACHE_KEY: ${{ secrets.BAZEL_CACHE_KEY }}
       WORKERS_MIRROR_URL: ${{ secrets.WORKERS_MIRROR_URL }}
+
+  workers-sdk-test:
+    needs: test
+    name: Run workers-sdk tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout workers-sdk
+        uses: actions/checkout@v4
+        with:
+          repository: cloudflare/workers-sdk
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+          cache: 'pnpm'
+      - name: Install workers-sdk dependencies
+        run: pnpm install
+
+      - name: Download workerd binary
+        uses: actions/download-artifact@v4
+        with:
+          name: linux-linux-64-binary
```

Comment:
Nit: This would need to be `linux-X64-binary` based on the binary artifact name as I understand. We should make it `linux-arm-ARM64-binary` though since the arm64 runner is faster. (The image for this job should be ubuntu-22.04-arm accordingly)